app/scripts/lib/relier-keys.js

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Derive relier-specific encryption keys from account master keys.
+ */
+
+'use strict';
+
+define([
+  'sjcl',
+  'p-promise',
+  'lib/hkdf',
+  'lib/constants'
+], function (sjcl, p, hkdf, Constants) {
+
+  /**
+   * Given the account master keys, the user id and the relier id, generate
+   * the matching relier-specific derived keys.
+   */
+  function deriveRelierKeys(keys, uid, rid) {
+    var relierKeys = {};
+    return p()
+      .then(function () {
+        if (! keys.kA) {
+          throw new Error('Cant derive relier keys: missing kA');
+        }
+        if (! keys.kB) {
+          throw new Error('Cant derive relier keys: missing kB');
+        }
+        return generateDerivedKey(keys.kA, 'kAr', uid, rid);
```

Comment:
We usually lift literals to the top of the scope and give them a descriptive variable name. This is more of a nit, but `generateDerivedKey` could also take an options object instead of four arguments– the property names would provide more context on the purpose of the values.